src/lava/magma/core/model/py/ports.py

```diff
@@ -9,96 +9,229 @@
 from lava.magma.compiler.channels.interfaces import AbstractCspPort
 from lava.magma.compiler.channels.pypychannel import CspSendPort, CspRecvPort
 from lava.magma.core.model.interfaces import AbstractPortImplementation
+from lava.magma.core.model.model import AbstractProcessModel
 from lava.magma.runtime.mgmt_token_enums import enum_to_np, enum_equal
 
 
 class AbstractPyPort(AbstractPortImplementation):
+    """Abstract class for Ports implemented in Python.
+
+    Ports at the Process level provide an interface to connect
+    Processes with each other. PyPorts provide an interface to send and
+    receive messages via channels implemented by a backend messaging
+    infrastructure. The Communicating Sequential Processes (CSP) paradigm is the
+    inspiration of the messaging infrastructure. Thus, a channel denotes a CSP
+    channel of the messaging infrastructure and CSP Ports denote the low level
+    ports also used in the messaging infrastructure. PyPorts are the
+    implementation for message exchange in Python, using the low level CSP Ports
+    of the backend messaging infrastructure. PyPorts provide interface for send
+    and receive of input and output Ports. A PyPort may have one or multiple
+    connection to other PyPorts. These connections are represented by csp_ports,
+    which is a list of CSP ports corresponding to the connected PyPorts.
+    """
     @property
     @abstractmethod
     def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
+        """
+        Abstract property to get the corresponding CSP Ports of all connected
+        PyPorts (csp_ports). The CSP Port is the low level interface of the
+        backend messaging infrastructure which is used to send and receive data.
+
+        Returns
+        -------
+        A list of all CSP Ports connected to the PyPort.
+        """
         pass
```

Comment:
What do you mean by "the corresponding CSP Ports of all connected PyPorts"?
Maybe "Abstract property of a PyPort that returns a list csp_ports of all CSP ports of PyPorts that this PyPort is connected to."?

Reply:
Only the corresponding CSP Ports used for the connection not all CSP Ports. Changed it to: "Abstract property to get a list of the corresponding CSP Ports of all connected PyPorts."